src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -168,6 +171,14 @@ private IEnumerable<string> GetViewNames()
 
                 yield return "String";
             }
+            else if (typeof(IFormFile).IsAssignableFrom(fieldType))
+            {
+                yield return nameof(IFormFile);
+            }
+            else if (typeof(IEnumerable<IFormFile>).IsAssignableFrom(fieldType))
+            {
+                yield return nameof(IFormFileCollection);
+            }
```

Comment:
also, this method _always_ yields "Collection", "String", or "Object" even if more-specific special cases exist in `_defaultEditorActions`.  that's required not just because `_defaultDisplayActions` contains many fewer keys (e.g. because depending on such information makes this method brittle).  don't break that invariant.